src/search.cpp

```diff
@@ -635,7 +635,8 @@ Value Search::Worker::search(
     // At non-PV nodes we check for an early TT cutoff
     if (!PvNode && !excludedMove && ttData.depth > depth - (ttData.value <= beta)
         && ttData.value != VALUE_NONE  // Can happen when !ttHit or when access race in probe()
-        && (ttData.bound & (ttData.value >= beta ? BOUND_LOWER : BOUND_UPPER)))
+        && (ttData.bound & (ttData.value >= beta ? BOUND_LOWER : BOUND_UPPER))
+        && (cutNode == (ttData.value >= beta) || depth > 8))
```

Comment:
where does the OR depth > 8 come from?

Reply:
The first implementation I tried didn't have this condition, but also didn't scale to LTC:
STC: https://tests.stockfishchess.org/tests/view/6721521586d5ee47d953d605
failed LTC: https://tests.stockfishchess.org/tests/view/672217ab86d5ee47d953d6c6
This suggested that for higher depths tt values are reliable enough even if `cutNode` doesn't agree.